prepare_abcd.R

```diff
@@ -0,0 +1,165 @@
+# set up project and load packages----
+library(dplyr, warn.conflicts = FALSE)
+library(readr)
+library(readxl)
+library(tidyr)
+
+# source helpers----
+source("expected_columns.R")
+
+# load config----
+config_dir <- config::get("directories")
+config_files <- config::get("file_names")
+
+path_abcd <- file.path(config_dir$dir_abcd, config_files$filename_abcd)
+sheet_abcd <- config_files$sheet_abcd
+
+config_prepare_abcd <- config::get("prepare_abcd")
+
+prepare_abcd_rm_inactive_companies <- config_prepare_abcd$remove_inactive_companies
+
+config_project_parameters <- config::get("project_parameters")
+
+project_parameters_start_year <- config_project_parameters$start_year
+project_parameters_time_frame <- config_project_parameters$time_frame
+
+# validate config values----
+if (!length(path_abcd) == 1) {
+  stop("Argument path_abcd must be of length 1. Please check your input.")
+}
+if (!inherits(path_abcd, "character")) {
+  stop("Argument path_abcd must be of class character. Please check your input.")
+}
+if (!length(sheet_abcd) == 1) {
+  stop("Argument sheet_abcd must be of length 1. Please check your input.")
+}
+if (!inherits(sheet_abcd, "character")) {
+  stop("Argument sheet_abcd must be of class character. Please check your input.")
+}
+if (!is.null(prepare_abcd_rm_inactive_companies)) {
+  if (!length(prepare_abcd_rm_inactive_companies) == 1) {
+    stop("Argument prepare_abcd_rm_inactive_companies must be of length 1. Please check your input.")
+  }
+  if (!inherits(prepare_abcd_rm_inactive_companies, "logical")) {
+    stop("Argument prepare_abcd_rm_inactive_companies must be of class logical. Please check your input.")
+  }
+}
+if (!length(project_parameters_start_year) == 1) {
+  stop("Argument project_parameters_start_year must be of length 1. Please check your input.")
+}
+if (!inherits(project_parameters_start_year, "integer")) {
+  stop("Argument project_parameters_start_year must be of class integer Please check your input.")
+}
+if (!length(project_parameters_time_frame) == 1) {
+  stop("Argument project_parameters_time_frame must be of length 1. Please check your input.")
+}
+if (!inherits(project_parameters_time_frame, "integer")) {
+  stop("Argument project_parameters_time_frame must be of class integer Please check your input.")
+}
+
+
+# load data----
+if (!file.exists(path_abcd)) {
+  stop(glue::glue("No ABCD file found at path {path_abcd}. Please check your project setup!"))
+}
+
+abcd <- readxl::read_xlsx(
+  path = file.path(path_abcd),
+  sheet = sheet_abcd,
+  col_types = cols_abcd$col_types_abcd
+)
```

Comment:
NB: I know @cjyetman has been using the `tidyxl` package lately. I'm not sure why/ if there is a benefit to that over `readxl`, but just flagging in case we want to align on an approach there.

Reply:
FYI... `tidyxl` allows you to read in the data and the formats of an XLSX file, so it's a bit more capable, though a bit more complicated to use. If there's a rectangular dataset in the XLSX you're importing, `readxl` will be easier.

My other reasoning for using it is it allows me to have only the importing of a raw XLSX file in a "workflow" and leave all the transformation, cell selection, variable classing, etc. for an R function that can fully specify what is done to the raw data. This is towards the goal of keeping all "methodology" stuff in a R package, and all/any file i/o stuff in a workflow.

So that's the context for it.

Reply:
okay, I think in this case it is fine to keep readxl then for now, as the banks abcd is always rectangular. But noted that it looks pretty useful for other datasets, looks like scenarios could be a perfect use case

Reply:
scenarios is exactly what I've been using it for 😉